dfm/cleaning/quality.py

```diff
@@ -0,0 +1,277 @@
+                two with the first element being the text. The output of this function
+                    will then also be a generator of tuples filtered based on the text.
+                    Defaults to False.
+
+        Yields:
+            Generator: A Generator of either texts or tuples depending on the as_tuples
+                argument.
+        """
+
+        for doc in self.nlp.pipe(docs, as_tuples=as_tuples):
+            if as_tuples:
+                doc, context = doc
+
+            valid = True
+            for filter, filter_fn in self.filters.items():
+                if not filter_fn(doc):
+                    valid = False
+
+                    # log filtered documents
+                    self.filtered[filter] += 1
+                    break
+            if not valid:
+                continue
+
+            if as_tuples:
+                yield doc, context
+            else:
+                yield doc
+
+    @staticmethod
+    def doc_length(doc: Doc, doc_length: Tuple[int, int]) -> bool:
+        """
+        A filter that removes any document that does not contain between {doc_length[0]}
+        and {doc_length[1]} words
+
+        Args:
+            doc (Doc): SpaCy document
+            doc_length (Tuple[int, int]): Tuple indicate minimum and maximum length of
+                document
+
+        Returns:
+            bool: A boolean indicator of whether the text passed the filter.
+        """
+
+        return doc_length[0] <= len(doc) <= doc_length[1]
+
+    @staticmethod
+    def mean_word_length(doc: Doc, mean_word_length: Tuple[int, int]) -> bool:
+        """
+        Filter document whose mean word length is outside the range of {mean_word_length[0]} to {mean_word_length[1]}
+        characters
+
+        Args:
+            doc (Doc): SpaCy document
+            mean_word_length (Tuple[int, int]): Tuple indicate minimum and maximum
+                mean word length of document
+
+        Returns:
+            bool: A boolean indicator of whether the text passed the filter.
+        """
+
+        w_len = [len(t) for t in doc]
+        mwl = sum(w_len) / len(doc)
+        return mean_word_length[0] <= mwl <= mean_word_length[1]
+
+    @staticmethod
+    def alpha(doc: Doc, ratio: float) -> bool:
+        """
+        Filter that requires the {ratio}% of words in a document contain at least one
+        alphabetic character
+
+        Args:
+            doc (Doc): SpaCy document
+            ratio (float): The ratio of alphabetic characters
+
+        Returns:
+            bool: A boolean indicator of whether the text passed the filter.
+        """
+        # checks if a non-space token contains a alphabetic character
+        contains_alpha = sum(
+            any(c.isalpha() for c in t.text) for t in doc if not t.is_space
+        )
+        len_doc = len(doc)
+        ratio_ = contains_alpha / len_doc
+        return ratio_ < ratio
```

Comment:
I think this should be the other way around: `ratio_ >= ratio`.

Reply:
Yep, should have run the test before pushing